### PR TITLE
Remove unnecessary crc32c implementation

### PR DESCRIPTION
**Pull request by MBkkt**
### Scope & Purpose

We used crc32c polynomial, not crc32 so we can use new abseil hardware accelerated implementation instead of our own.

Btw if sometime you will needs crc32 please use folly or boost.

- [ ] :hankey: Bugfix
- [ ] :pizza: New feature
- [ ] :fire: Performance improvement
- [ ] :hammer: Refactoring/simplification

### Checklist

- [ ] Tests
  - [ ] **Regression tests**
  - [ ] C++ **Unit tests**
  - [ ] **integration tests**
  - [ ] **resilience tests**
- [ ] :book: CHANGELOG entry made
- [ ] :books: documentation written (release notes, API changes, ...)
- [ ] Backports
  - [ ] Backport for 3.10: *(Please link PR)*
  - [ ] Backport for 3.9: *(Please link PR)*
  - [ ] Backport for 3.8: *(Please link PR)*

#### Related Information

*(Please reference tickets / specification / other PRs etc)*

- [ ] Docs PR: 
- [ ] Enterprise PR: https://github.com/arangodb/enterprise/pull/1155
- [ ] GitHub issue / Jira ticket:
- [ ] Design document:

### CODE REVIEW

File: tests/Basics/HashesTest.cpp [hunk 1]
```diff
@@ -159,156 +185,156 @@ TEST_F(CHashesTest, tst_fnv64_simple) {
...
-  EXPECT_TRUE(14695981039346656037ULL ==
-              TRI_FnvHashPointer(buffer.c_str(), strlen(buffer.c_str())));
+  EXPECT_TRUE(14695981039346656037ULL == FnvHashString(buffer.c_str()));
   EXPECT_TRUE(14695981039346656037ULL ==
-              TRI_FnvHashBlock(TRI_FnvHashBlockInitial(), buffer.c_str(),
-                               strlen(buffer.c_str())));
+              FnvHashPointer(buffer.c_str(), strlen(buffer.c_str())));
+  EXPECT_TRUE(14695981039346656037ULL == FnvHashBlock(kFnvHashBlockInitial,
+                                                      buffer.c_str(),
+                                                      strlen(buffer.c_str())));
```
Comment (jsteemann) [inline tests/Basics/HashesTest.cpp:193]: is it possible to replace all occurences of `strlen(buffer.c_str())` with just `buffer.size()`?

File: tests/Basics/HashesTest.cpp [hunk 2]
```diff
@@ -43,8 +45,32 @@ class CHashesTest : public ::testing::Test {
   CHashesTest() {
     IcuInitializer::setup("./3rdParty/V8/v8/third_party/icu/common/icudtl.dat");
   }
+
+  uint32_t testCrc32(std::string& buffer) {
```
Comment (jsteemann) [inline tests/Basics/HashesTest.cpp:49]:
```suggestion
  uint32_t testCrc32(std::string const& buffer) {
```

File: tests/Basics/HashesTest.cpp (hunk 1 above)
Comment (MBkkt) [reply tests/Basics/HashesTest.cpp:193]: I think yes. honestly because of it in test, I just decide don't touch it

**Review by jsteemann — COMMENTED** [review]
Does this PR also allow us to get rid of the following files (and their CMake integration)?
* lib/Basics/crc1.S  
* lib/Basics/crc4.S  
* lib/Basics/crc5.S

**Review by jsteemann — APPROVED** [review]
LGTM

**Comment by MBkkt** [comment]
https://jenkins01.arangodb.biz/job/arangodb-matrix-pr/25880/

**Pull request merged**